### DISCUSSION TITLE
NEWS:  Moving to CommunityToolkit and the dotNet Foundation

### DISCUSSION THREAD

**Discussion by adrianhall** [Azure/azure-mobile-apps#912] [General]
Hey folks,

I wanted to provide some more details on the proposed move to CommunityToolkit that was filed in issues this morning.

For some time, we've been caught in limbo for releases.  The 6.1.1 release has not come out yet because I have some more hoops to jump through to ensure secure software supply-chain and signing requirements are met (which I'm working on).  Long term, this is only going to get worse because nation actors are attacking Microsoft constantly, and that - in turn - is getting Microsoft to lock down which projects get signed as "belonging to Microsoft"

Azure Mobile Apps is currently community driven, and is not support by Microsoft (except via me), so that combined with the continuing lock down to secure the software supply chain, is causing continued engineering systems / release issues.

I've reach an agreement in principal to move the project to the Community Toolkit.  If you have never heard of the Community Toolkit, you can check it out here: https://github.com/CommunityToolkit  and the docs are here: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/introduction  As an organization, they provide controls and libraries for building Windows, MAUI, Xamarin applications.  It includes things like an MVVM framework (which is awesome), 

Most importantly, Community Toolkit is a part of the dotNet foundation (DNF).  DNF provides signing capabilities for us so we can run our own engineering systems yet still be trusted as an open source project.  I believe it will also allow more people to contribute to the code by taking it out from under the Azure/Microsoft organization.

This doesn't change any of the plans I have for the library.  Indeed, I'll be able to post more of the roadmap and work more openly on the move to EFCore that we have planned in the client.  So stay tuned - this is all positive moves!  The only thing I can't answer right now is "when" - I'm still gathering all the written agreements I need internally and with the dotNet Foundation to make the move.  Once those approvals are in place, the move will happen pretty quickly.

Feel free to ask any questions below and I'll do my best to answer them quickly.

**Comment by adrianhall** [comment]
Hey folks,

I wanted to provide an update here.  I can no longer publish new packages in the "Azure" organization.  

The new repository is up and running.  However, I'm working on the new client.  As a reminder, the new client will be DbContext / Entity Framework Core based, so it will take some time to get published.

I am happy to answer questions on Azure Mobile Apps here (and will continue to do so).  Once we release the Community Toolkit edition, this repository will go into read-only mode pretty quickly.

**Reply by hansmbakker** [reply]
Hi, I see the issues section disappeared - was that intentional?

Other than that, thank you for the update!

**Reply by adrianhall** [reply]
Yes - that was intentional.  I will not be working issues in this repository any more (since I cannot release any updates I do make).

The "new" repository contains all the code necessary to build the server (with the exception of the NSwag and Swashbuckle stuff).  The client is actually a ton of new code (and you are quite welcome to contribute with code reviews, discussions, questions, etc. once I have checked in the client).

**Reply by hansmbakker** [reply]
Would it be possible for you to move the issue I reported to the discussions? Because the issues section is removed, I cannot read the answer you gave on my issue.

**Reply by adrianhall** [reply]
I basically agreed with you.  However, since a lot of the sync logic is being re-written for the community toolkit, the bug you found won't be worth dragging over to the new codebase.

**Comment by anpin** [comment]
here is the link to the new repo https://github.com/CommunityToolkit/Datasync

**Comment by anpin** [comment]
@adrianhall would you keep compatibility of the new communitytoolkit version with the server found in this repo? 

We just finished migration from Xamarin.Forms to MAUI, including transitioning from v4.x to v6.x of the Datasync library and it was rather painful. Having to do it all over again is simply not feasible.

**Reply by adrianhall** [reply]
I'm working on cross-compatibility.  The server is mostly staying the same (we are dropping a lot of the "ZUMO" isms such as the required protocol header).  The v6 client should work with the community toolkit server.  The reverse will also be true, but will require an extra optional plug-in.

You can try this today as the server is pretty much complete.  Just download the source from github.com/CommunityToolkit/Datasync.

**Reply by anpin** [reply]
@adrianhall  Is it possible to return the issue section on github, there was a useful information, but now it is not accessible anymore

**Reply by adrianhall** [reply]
No, it isn't.

**Comment by adrianhall** [comment]
Hey folks - quick update.

The server section on CommunityToolkit.Datasync is done - however, the client is causing me no end of issues, with no clear resolution timeline. 

The idea, if you remember, was to center the design around Entity Framework Core on the client.  it turns out thats really hard to do generically.  I've started and reset three times now, so I have no updated timeline to report other than "I'm working on it"
